### ISSUE TITLE
White space BBOX is Wrong

### ISSUE THREAD

**Issue by mailsnathaniel** [pymupdf/PyMuPDF#823]
Hi,

I White space bbox is wrong. I have even used ascender/decender to get the actual ymin and ymax. 

I have attached the input and output (span chunks are marked in red outline).

FYI - This input pdf is created using ABBY OCR.

Configurations:
 - Ubuntu
 - Python3.6
 - PyMuPDF 1.18.6
 
Thanks
![spaces_bbox](https://user-images.githubusercontent.com/64656539/104268312-9dea6800-54b9-11eb-966e-6270b0db1532.jpg)
[cheesecake-20191221_003.pdf](https://github.com/pymupdf/PyMuPDF/files/5799731/cheesecake-20191221_003.pdf)

**Comment by JorjMcKie** [comment]
Found the reason:
As you know, the fontsize of a span plays a pivotal role when creating the small span bboxes. MuPDF returns me large fontsize values in case of most of the spans that have spces in their text, see for example here for the very large bbox after "Sub Total:"
```python
      {
       "size":10.201054573059082,
       "flags":0,
       "font":"ArialMT",
       "color":0,
       "ascender":1.0750000476837158,
       "descender":-0.29899999499320984,
       "text":"Total:",
       "origin":[
        129.0,
        228.2010040283203
       ],
       "bbox":[
        129.0,
        217.96078491210938,
        156.18267822265625,
        231.04920959472656
       ]
      },
      {
       "size":29.90174102783203,  # <===  look at this! Compare to bbox height!
       "flags":0,
       "font":"ArialMT",
       "color":0,
       "ascender":1.0750000476837158,
       "descender":-0.29899999499320984,
       "text":" ",
       "origin":[
        156.0,
        228.2010040283203
       ],
       "bbox":[  # but the bbox height is just over 13, less than 50% of fontsize!
        156.0,
        217.98611450195312,
        182.1584930419922,
        231.0421600341797
       ]
      },
```
Whatever the reason for this may be: all you can do is rejecting the "reduced" span bbox if it has no smaller height than the original.
I will adjust the recipe in the documentation accordingly, and also the code behind `fitz.TOOLS.set_small_bbox_heights()`.

**Comment by JorjMcKie** [comment]
Once this is done, the result will look like this, and that's the end of it:
![grafik](https://user-images.githubusercontent.com/8290722/104297806-a8950500-5499-11eb-91a3-823ed57eeebb.png)

**Comment by JorjMcKie** [comment]
Actually, this ridiculous fontsize of 29.9 is contained in the PDF - so ABBYY is to blame, not (Py-) MuPDF.

You even must say, that it is a pretty good job to come up with a reasonable bbox under these circumstances!

**Comment by JorjMcKie** [comment]
Fixed by v1.18.7, currently being uploaded.

**Comment by mailsnathaniel** [comment]
Thanks.
